app/models/listing_shape.rb

```diff
@@ -35,6 +35,8 @@ class ListingShape < ActiveRecord::Base
     :price_quantity_placeholder,
   )
 
+  validates_uniqueness_of :name, :scope => :community_id
```

Comment:
Is this validated with unique index? This type of model validation if quite slow in my understanding so it might be good enough to leave it to db as long as we're making serious effort to generate a unique name when saving shape.